src/proposals/20250109/SparkEthereum_20250109.sol

```diff
@@ -0,0 +1,397 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.25;
+
+import {
+    Ethereum,
+    Base,
+    SparkPayloadEthereum,
+    IEngine,
+    EngineFlags,
+    Rates
+} from "../../SparkPayloadEthereum.sol";
+
+import { IERC20 } from "forge-std/interfaces/IERC20.sol";
+
+import { Base } from "spark-address-registry/Base.sol";
+
+import { CCTPForwarder } from "xchain-helpers/forwarders/CCTPForwarder.sol";
+
+import { IMetaMorpho, MarketParams } from "metamorpho/interfaces/IMetaMorpho.sol";
+
+import { AllocatorBuffer } from "dss-allocator/src/AllocatorBuffer.sol";
+import { AllocatorVault }  from "dss-allocator/src/AllocatorVault.sol";
+
+import { ControllerInstance }              from "spark-alm-controller/deploy/ControllerInstance.sol";
+import { MainnetControllerInit }           from "spark-alm-controller/deploy/MainnetControllerInit.sol";
+import { MainnetController }               from "spark-alm-controller/src/MainnetController.sol";
+import { RateLimitHelpers, RateLimitData } from "spark-alm-controller/src/RateLimitHelpers.sol";
+
+import { ICapAutomator } from "sparklend-cap-automator/interfaces/ICapAutomator.sol";
+
+interface ITokenBridge {
+    function bridgeERC20To(
+        address _localToken,
+        address _remoteToken,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    ) external;
+}
+
+/**
+ * @title  Jan 09, 2025 Spark Ethereum Proposal
+ * @notice Sparklend: update WBTC, cbBTC, wstETH, weETH, DAI, USDC and USDT parameters
+ *         Morpho: onboard PT-USDe-29May2025 and adjust existing caps
+           Spark Liquidity Layer: Controller Upgrade, Onboard Ethena Direct,
+                                  Aave V3 and increase liquidity available to Base
+ * @author Phoenix Labs
+ * Forum:  https://forum.sky.money/t/27-dec-2024-proposed-changes-to-spark-for-upcoming-spell/25760
+ * Vote:   TODO
```

Comment:
https://vote.makerdao.com/polling/QmYScEHT
https://vote.makerdao.com/polling/Qma1xA18
https://vote.makerdao.com/polling/QmeEvuG2
https://vote.makerdao.com/polling/QmVpv1G8
https://vote.makerdao.com/polling/QmZ2Qmy5
https://vote.makerdao.com/polling/QmSTYyW5
https://vote.makerdao.com/polling/QmTFsGw8
https://vote.makerdao.com/polling/QmYDGkjM
https://vote.makerdao.com/polling/QmZfSDMH
https://vote.makerdao.com/polling/QmQ6bYou (moving funds to base)

Reply:
added polls